frappe/database/database.py

```diff
@@ -260,6 +264,24 @@ def sql_ddl(self, query, values=(), debug=False):
 		self.commit()
 		self.sql(query, debug=debug)
 
+	@staticmethod
+	def check_permissions(doctype, **kwargs):
+		kwargs.pop("ignore_permissions")
+		if isinstance(doctype, str):
+			doctype = [doctype]
+
+		for dt in doctype:
+			dt = re.sub("tab", "", dt)
+			if not frappe.has_permission(
+				dt, "select", **kwargs
+			) and not frappe.has_permission(dt, "read", **kwargs):
+				frappe.flags.error_message = _('Insufficient Permission for {0}').format(frappe.bold(dt))
+				raise frappe.PermissionError(dt)
```

Comment:
This could be a `frappe.throw`?

Reply:
Raised was used in db_query thus I used it here, didn't know if I should use throw or raise

Reply:
Use throw for consistency. Also, you won't have to manually set error_message flag.

Reply:
refactored in 68fe869